card_records.rb

```diff
@@ -0,0 +1,219 @@
+require "open-uri"
+
+module NewMexico
+  class CardRecords
+    HOST = "https://wwwapps.emnrd.state.nm.us/"
+    PERMIT_PATH = "ocd/ocdpermitting/OperatorData/PermitStatusParameters.aspx"
+    AWS_BUCKET_PATH = "pdocument/well_documents/nm/permit_documents/"
+    RENAME = {
+      "Id" => "filing_id",
+      "Type" => "document_type",
+      "Description" => "document_comment",
+      "Status" => "document_status",
+      "Status Date" =>"scan_date"
+    }.freeze
+
+    RENAME2 = {
+      "Permit:" => "permit_id",
+      "Image Date:" => "scan_date",
+      "Operator:" => "operator_name",
+      "Current Operator:" => "operator_name",
+      "Well Name & Number:" => "well_name_full",
+      "Well Name and Number:" => "well_name_full",
+      "Well Info:" => "well_name_full",
+      "ULSTR:" => "location",
+      "Previous Name:" => "previous_name",
+      "New Name:" => "new_name",
+      "New Operator:" => "new_name",
+      "Effective Date:" => "effective_date"
+    }.freeze
+
+    def run
+      goto(uri)
+      fill_form
+      submit_form
+
+      browser.wait_until { |b| b.table(id: "ctl00_ctl00__main_main_gvResults").exists? }
+
+      Array.new(count_of_pages) { |i| (i + 1).to_s }.each do |page_number|
+        link = pagination.link(text: page_number)
+
+        link.click unless link.class_name == "active"
+
+        browser.wait_until { |b| pagination.link(text: page_number).class_name == "active" }
+
+        rows.each do |row|
+          query_params = row_params(row)
+          query_params["scan_date"] = strpdate(query_params["scan_date"])
+          query_params["well_card_url"] = row.link.href
+
+          NewMexicoQueryRecord.new(query_params).save
+
+          browser.execute_script("window.open(\"#{query_params["well_card_url"]}\")")
```

Comment:
To make sure I understand, this will open a new window for every card, then you access it below by operating on the "last window" `browser.windows.last`?

Reply:
Yes, this is because this site, after clicking the "Go Back" button, returns to the first page.